src/osemgrep/cli_scan/Core_runner.ml

```diff
@@ -138,53 +110,14 @@ let group_rules_by_target_language rules : (Xlang.t * Rule.t list) list =
   Hashtbl.fold (fun lang rules acc -> (lang, rules) :: acc) tbl []
 
 let split_jobs_by_language all_rules all_targets : Lang_job.t list =
-  let grouped_rules = group_rules_by_target_language all_rules in
-  let cache = Filter_target.create_cache () in
-  grouped_rules
-  |> Common.map_filter (fun (lang, rules) ->
-         let rules, targets =
-           match lang with
-           | Xlang.LGeneric
-           | Xlang.LRegex ->
-               let rules, targets =
-                 List.fold_left
-                   (fun (rules, targets) rule ->
-                     let required_path_patterns, excluded_path_patterns =
-                       match rule.Rule.paths with
-                       | Some { include_; exclude } -> (include_, exclude)
-                       | None -> ([], [])
-                     in
-                     let targets' =
-                       all_targets
-                       |> List.filter (fun target ->
-                              Filter_target.filter_target_for_lang ~cache ~lang
-                                ~required_path_patterns ~excluded_path_patterns
-                                target)
-                     in
-                     if Common.null targets' then (rules, targets)
-                     else (rule :: rules, targets @ targets'))
-                   ([], []) rules
-               in
-               (rules, List.sort_uniq Fpath.compare targets)
-           | _else ->
-               ( rules,
-                 all_targets
-                 |> List.filter (fun target ->
-                        rules
-                        |> List.exists (fun (rule : Rule.t) ->
-                               let ( required_path_patterns,
-                                     excluded_path_patterns ) =
-                                 match rule.paths with
-                                 | Some { include_; exclude } ->
-                                     (include_, exclude)
-                                 | None -> ([], [])
-                               in
-                               Filter_target.filter_target_for_lang ~cache ~lang
-                                 ~required_path_patterns ~excluded_path_patterns
-                                 target)) )
+  all_rules |> group_rules_by_target_language
+  |> Common.map_filter (fun (xlang, rules) ->
+         let targets =
+           all_targets
+           |> List.filter (Filter_target.filter_target_for_xlang xlang)
          in
-         if Common.null rules || Common.null targets then None
-         else Some ({ lang; targets; rules } : Lang_job.t))
+         if Common.null targets then None
+         else Some ({ xlang; targets; rules } : Lang_job.t))
```

Comment:
the previous code, for Generic and Regex rules, did as well filter the rules (for which apply / do not apply -- lines 149 -- 168) -- but I guess this can be done differently / separately. Also, the paths include/exclude of the rules aren't take into constderation anymore at all...

Reply:
Yes, I'll make a PR to handle include/exclude, but I'll do it in a different place. This whole List.exists was not working anyway.